tutorials/3d/portals/fundamentals.rst

```diff
@@ -0,0 +1,102 @@
+
+Room naming convention
+^^^^^^^^^^^^^^^^^^^^^^
+Unlike most nodes in Godot, a specific naming convention should be followed in order to identify each room. The name should have the prefix ``Room_`` followed by the name you wish to give the room, e.g. ``Room_kitchen``, ``Room_lounge``. If you don't follow these naming guidelines, the system will warn you and may not work correctly.
+
+How do I define the shape and position of my room convex hull?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Because defining the room bound is the most important aspect of the system, there are THREE methods available to define the shape of a room in Godot:
+
+1. Use the geometry of the objects contained within the room to automatically create an approximate bound
+2. Provide a manual bound - a MeshInstance in the room that has geometry in the shape of the desired bound, with a name prefixed by ``Bound_``. This is something you might choose to do if you create your levels in Blender or similar (see :ref:`doc_rooms_and_portals_blender`).
+3. By manually editing the points that define the convex hull, in the room inspector.
+
+While the first option can be all that is required, particularly with simple rooms, or for pre-production, the power of the manual bounds gives you ultimate control (at the expense of a small amount of editing). You can also combine the two approaches, perhaps using automatic bounds for most rooms but manually editing problem areas.
+
+The automatic method is used whenever a manual bound is not supplied.
+
+*A simple pair of rooms. The portal margin is shown with translucent red, and the room hulls are shown with green wireframe.*
+
+.. image:: img/simple_room.png
+
+Portals
+~~~~~~~
+If you create some rooms, place objects within them, then convert the level in the editor, you will see the objects in the rooms appearing and showing as you move between rooms. There is one problem however! Although you can see the objects within the room that the camera is in, you can't see to any neighbouring rooms! For that we need portals.
+
+:ref:`Portal<class_Portal>`\ s are special convex polygons. You position over the openings between rooms, in order to allow the system to see between them. You can create a portal node directly in the editor. The default portal has 4 points and behaves much like a ``plane`` :ref:`MeshInstance<class_MeshInstance>`. You can add or remove points using the inspector. A portal will require at least 3 points to work (in order to form a polygon rather than a point or line).
+
+Portals only need to be placed in one of each pair of neighbouring rooms (the *'source room'*) - the system will automatically make them two way unless you choose otherwise in the portal settings. The portal normal should face *outward* from the source room. The front face should be visible from *outside* the room. The editor gizmo indicates the direction the portal is facing with an arrow, and a different color for each face.
```

Comment:
```suggestion
Portals only need to be placed in one of each pair of neighbouring rooms (the *"source room"*). The system will automatically make them two way unless you choose otherwise in the portal node's properties. The portal normal should face *outward* from the source room. The front face should be visible from *outside* the room. The editor gizmo indicates the direction the portal is facing with an arrow, and a different color for each face.
```